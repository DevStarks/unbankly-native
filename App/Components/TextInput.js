import React, { Component } from 'react'
import { TextInput, View } from 'react-native'
import styles from './Styles/TextInputStyles'
import ExamplesRegistry from '../Services/ExamplesRegistry'

// Ignore in coverage report
/* istanbul ignore next */

ExamplesRegistry.addComponentExample('TextInput', () =>
  <AppTextInput>This is text</AppTextInput>
)

class AppTextInput extends Component {
  render () {
    return (
      <View>
        <TextInput {...this.props} style={[styles.textInput, this.props.style]}>
          {this.props.children}
        </TextInput>
        <View style={styles.textInputUnderline} />
      </View>
    )
  }
}

export default AppTextInput
