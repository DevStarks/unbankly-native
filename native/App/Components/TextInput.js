import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { TextInput, View } from 'react-native'
import styles from './Styles/TextInputStyles'
import ExamplesRegistry from '../Services/ExamplesRegistry'

// Ignore in coverage report
/* istanbul ignore next */

ExamplesRegistry.addComponentExample('TextInput', () =>
  <AppTextInput>This is text</AppTextInput>
)

class AppTextInput extends Component {
  // For a complete list of props
  // https://facebook.github.io/react-native/docs/textinput.html#props

  static propTypes = {
    onChangeText: PropTypes.func.isRequired
  }

  render () {
    return (
      <View style={[this.props.style, styles.inputWrap]}>
        <TextInput {...this.props} style={styles.textInput}>
          {this.props.children}
        </TextInput>
        <View style={styles.textInputUnderline} />
      </View>
    )
  }
}

export default AppTextInput
