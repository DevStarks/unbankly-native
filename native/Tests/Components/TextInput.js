import 'react-native'
import React from 'react'
import TextInput from '../../App/Components/TextInput'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

test('TextInput component renders correctly', () => {
  const tree = renderer.create(<TextInput onChangeText={() => {}} text='hi' />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onChangeText', () => {
  let i = '' // i guess i could have used sinon here too... less is more i guess
  const onChangeText = () => { i = 'changed' }
  const wrapperPress = shallow(<TextInput onChangeText={onChangeText} />)

  expect(wrapperPress.prop('onChangeText')).toBe(onChangeText) // uses the right handler
  expect(i).toBe('')
  wrapperPress.simulate('changeText')
  expect(i).toBe('changed')
})
