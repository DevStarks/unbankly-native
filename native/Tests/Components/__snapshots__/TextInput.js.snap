// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "#E7E7E7",
        "borderWidth": 1,
        "color": "#637381",
        "fontFamily": "AmaticSC-Bold",
        "fontSize": 20,
        "height": 60,
        "paddingBottom": 16,
        "paddingLeft": 5,
        "paddingTop": 16,
      }
    }
    text="hi"
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "backgroundColor": "#ECC925",
        "bottom": 4,
        "height": 1,
        "left": 1,
        "width": 291,
      }
    }
  />
</View>
`;
